### ISSUE TITLE
Question about waypoint_flier_simple

### ISSUE THREAD

**Issue by manueldiaz96** [ctu-mrs/example_ros_packages#2]
Hello, I was able to properly install the Apptainer image, but when moving to trying to run the `mrs_core_examples/waypoint_flier_simple`  even though Rviz launches, the terminal executing the waypoint flyer gets stuck at:
```
[ INFO] [1721158341.194518470, 1721158341.157600164]: [WaypointFlierSimple]: Waiting  for activation
```
And looking at the code, this would only happen id the `WaypointFlierSimple::onInit()` method does not run. Aside from just launching the script under `mrs_core_examples/cpp/waypoint_flier_simple/tmux/start.sh` , do I need to launch anything else?

What I expected when running the start script was to see the drone in Rviz, start moving around to random points, and to see the corresponding ROS INFO messages: 

```
[WaypointFlierSimple]: Distance to reference:   dist_to_ref

[WaypointFlierSimple]: Goal reached!

[WaypointFlierSimple]: New goal X:  goal_x_  Y: goal_y_  Z: << goal_z_
```

Is what I am expecting wrong? 
Do I need to also launch Gazebo by any chance?

Thanks for the help!

**Comment by cturmrsbot** [comment]
Hey, the example works as intended, it is not supposed to do anything more than having the drone flying around to random points. The `onInit()` function runs, since it starts the timer and initializes the subscribers / publishers / service clients, which obviously work. What makes you think it does not run?

The simulation session does not utilize Gazebo, but our minimalistic dynamic simulation. The only visual output is happening through the Rviz.

The purpose of the example package is to provide a "reference" from which people can copy pieces of code into their own nodes. Besides, it showcases how to implement some plugins into our system.

**Comment by manueldiaz96** [comment]
It makes me think that it does not run since the node does not output any other message than:
```
[ INFO] [1721158341.194518470, 1721158341.157600164]: [WaypointFlierSimple]: Waiting  for activation
```

And the drone does not move in the Rviz visualization at all, even after waiting for 10 minutes after launching `./tmux/start.sh`

**Comment by cturmrsbot** [comment]
Ok, will check that.

**Comment by cturmrsbot** [comment]
Here is a video how on works on my machine: https://youtu.be/jgQ9C_q_Tpw

**Comment by manueldiaz96** [comment]
Thanks for the video!

I will try and record a video later today (I have the MRS system installed in my personal PC) to see what the problem is. 

Do you have any ideas on what other info I can provide together with the video to help debug this problem?

**Comment by cturmrsbot** [comment]
The video will be a good start. Definitely check what is the result of calling the service? Did it get throught? Did it output any error? The `onInit()` definitely runs, otherwise it would not print "waiting for actication". But if you want to be sure, just add some print at the end of the function.

**Comment by manueldiaz96** [comment]
I did notice that you called `rosservice start /uav1/waypoint_flyer_simple/start`

Did you run the command yourself or is it run by the `start.sh` script? 

Because looking at the yaml file, I see this line: 

https://github.com/ctu-mrs/mrs_core_examples/blob/f3808c6ebfbbce7513bee486a06e670604aafeb3/cpp/waypoint_flier_simple/tmux/session.yml#L44

Does that line mean that the command runs by itself? 
Or does it mean that it gets put into the history, so it is easier to execute by pressing `UpArrowKey` and then `Enter`?

**Comment by cturmrsbot** [comment]
`history -s ` pushes the command into the shell's history, making it easy to run manually, as you written. 

Ye, this is an example, the service call is arbitrary and it is there just to give the developers an example for how to write a service server.

**Comment by klaxalk** [comment]
Aah sorry for answering as out bot account, forgot to switch.

**Comment by manueldiaz96** [comment]
Tested and worked! 

The error was in not knowing that I needed to launch the command for the service on the other terminal as I said here https://github.com/ctu-mrs/mrs_core_examples/issues/2#issuecomment-2232744811

In this pull request, I proposed some changes to the README of the WaypointFlyer and WaypointFlyerSimple to make it more explicit for other people to understand how to launch the services. 

[Link to Pull Request #3](https://github.com/ctu-mrs/mrs_core_examples/pull/3)

I hope this contribution is valuable for the newbie like me who is interested in tinkering a bit with these examples, but does not fully know how to launch the script/code. Feel free to reject it if you think it does not add much to the repo.

Thanks again!
